components/order-menu/main-menu/index.js

```diff
@@ -1,10 +1,18 @@
-export default function Maindishes() {
+import { Product } from "@/components/Product";
+import { maindishes } from "@/pages/_app.js";
+export default function Maindishes({ handleAddItem, items }) {
   return (
     <ul>
-      <li>Schnitzel</li>
-      <li>Spätzle</li>
-      <li>Rindergulasch</li>
-      <li>Sauerbraten</li>
+      {maindishes.map((maindish) => (
+        <Product
+          handleAddItem={handleAddItem}
+          key={maindish.id}
+          id={maindish.id}
+          name={maindish.name}
+          price={maindish.price}
+          items={items}
+        />
+      ))}
```

Comment:
Good! Here you'll have to pass the amount like:

```js
<Product
  handleAmountChange={handleAmountChange}
  key={maindish.id}
  amount={items.find((item) => item.id === maindish.id)?.amount}
  {...maindish}
/>
```

`items.find((item) => item.id === maindish.id)?.amount` finds the amount from the items (which I would call order or cart).

The `?.` is to save yourself a ternary in case there is no amount yet.

The `items` should not be passed now, as the amount consideration is done outside from the `Product`!